src/parser/validator.ts

```diff
@@ -0,0 +1,52 @@
+import { RootParser } from "../parser/RootParser";
+import { makeCstVisitor, VisitorResult, CommandNode } from "../parser/makeCstVisitor";
+
+interface validateCSTResult {
```

Comment:
Please use this style: `CstResult` (abbreviations are camel-cased).

Also, interface names should start with a capital letter.